integration/src/integration/java/pl/allegro/tech/hermes/integration/FrontendAuthenticationConfigurationTest.java

```diff
@@ -104,13 +105,49 @@ public void shouldNotAuthenticateUserWithInvalidCredentials() throws Throwable {
     @Test
     public void shouldNotAuthenticateUserWithoutCredentials() throws Throwable {
         //when
-        Response response = publisher.publish("someGroup.topicWithAuthorization",
-                TestMessage.of("hello", "world").body());
+        Response response = publisher.publish("someGroup.topicWithAuthorization", MESSAGE);
 
         //then
         assertThat(response.getStatus()).isEqualTo(StatusCodes.UNAUTHORIZED);
     }
 
+    @Test
+    public void shouldPublishOnTopicWithGrantedPermission() {
+        //given
+        Topic topic = TopicBuilder.topic("g.granted")
+                .withPublisher(USERNAME)
+                .withAuthEnabled()
+                .withAuthEnforced()
+                .build();
+
+        operations.buildTopic(topic);
+        Map<String, String> headers = getHeadersWithAuthentication(USERNAME, PASSWORD);
+
+        //when
+        Response response = publisher.publish(topic.getQualifiedName(), MESSAGE, headers);
+
+        //then
+        assertThat(response.getStatusInfo().getFamily()).isEqualTo(SUCCESSFUL);
+    }
+
+    @Test
+    public void shouldNotPublishOnTopicWithoutGrantedPermission() {
```

Comment:
I think we should add tests for cases where:

- topic publishers list is not empty  & auth is enabled but not enforced (like during migration) and:
  - the logged in user is not in publishers list (say we have an error in topic config) (should be 403)
  - the user is not logged in (should be 200)

---
What's more, how about below cases just to describe the logic/behaviour, as this may be a little confusing (they are possible, but may sound like nonsense from our perspective) - what do you think?
- topic publishers list is empty & auth is enabled but not enforced:
  -  the user is logged in (403?)
  - the user is not logged in (200?)

- topic publishers list is empty & auth is enabled and enforced:
  - the user is logged in (403, we have test for this)
  - the user is not logged in (403?)